src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -313,6 +313,92 @@ impl ArithmeticOp for DecimalMultiply {
     }
 }
 
+#[derive(Debug)]
+pub struct IntDivideInt;
+
+impl ArithmeticOp for IntDivideInt {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(div_i64(*lhs, *rhs)?))
+    }
+}
+
+#[derive(Debug)]
+pub struct IntDivideUint;
+
+impl ArithmeticOp for IntDivideUint {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(div_i64_with_u64(*lhs, *rhs as u64).map(|r| r as i64)?))
+    }
+}
+
+#[derive(Debug)]
+pub struct UintDivideUint;
+
+impl ArithmeticOp for UintDivideUint {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(((*lhs as u64) / (*rhs as u64)) as i64))
+    }
+}
+
+#[derive(Debug)]
+pub struct UintDivideInt;
+
+impl ArithmeticOp for UintDivideInt {
+    type T = Int;
+
+    fn calc(lhs: &Int, rhs: &Int) -> Result<Option<Int>> {
+        if *rhs == 0 {
+            return Ok(None);
+        }
+        Ok(Some(div_u64_with_i64(*lhs as u64, *rhs).map(|r| r as i64)?))
+    }
+}
+
+#[derive(Debug)]
+pub struct IntDivideDecimal;
+
+impl ArithmeticOp for IntDivideDecimal {
+    type T = Decimal;
+
+    fn calc(lhs: &Decimal, rhs: &Decimal) -> Result<Option<Decimal>> {
+        use crate::coprocessor::codec::mysql::Res;
+
+        let overflow = Error::overflow("DECIMAL", &format!("({} / {})", lhs, rhs));
+
+        let s: Result<Option<Decimal>> = match lhs / rhs {
+            Some(v) => match v {
+                Res::Ok(v) => match v.as_i64() {
+                    Res::Ok(v_i64) => Ok(Some(Decimal::from_f64(v_i64 as f64).unwrap())),
+                    Res::Truncated(v_i64) => Ok(Some(Decimal::from_f64(v_i64 as f64).unwrap())),
+                    Res::Overflow(_) => {
+                        Err(Error::overflow("BIGINT", &format!("({} / {})", lhs, rhs)))?
+                    }
+                },
+                Res::Truncated(_) => Err(Error::truncated())?,
+                Res::Overflow(_) => Err(overflow)?,
+            },
+            None => Err(Error::division_by_zero())?,
```

Comment:
Original implementation has `handle_division_by_zero`.

Reply:
> Original implementation has `ctx.handle_division_by_zero`.

`ctx.handle_division_by_zero` will check whether the database is using strict mode, add `division_by_zero` message to warning, and return `None` or `Error(division_by_zero)`. How can I get these infomation in rpn_expr? if we cannot get these infomation in `rpn_expr`, maybe just return `None` is OK?